# Generated from https://pypi.org/project/asyncssh/
package:
  name: py3-asyncssh
  version: "2.21.1"
  epoch: 0
  description: AsyncSSH is a Python package which provides an asynchronous client and server implementation of the SSHv2 protocol
  copyright:
    - license: EPL-2.0 OR GPL-2.0-or-later

vars:
  pypi-package: asyncssh

data:
  - name: py-versions
    items:
      3.10: '310'
      3.11: '311'
      3.12: '312'
      3.13: '313'

environment:
  contents:
    packages:
      - build-base
      - busybox
      - ca-certificates-bundle
      - py3-supported-pip
      - py3-supported-python-dev
      - py3-supported-setuptools
      - wolfi-base

pipeline:
  - uses: git-checkout
    with:
      repository: https://github.com/ronf/asyncssh
      tag: v${{package.version}}
      expected-commit: ec588e4e8145b6d7d06af82e081c9ebafa03eec2

subpackages:
  - range: py-versions
    name: py${{range.key}}-${{vars.pypi-package}}
    description: python${{range.key}} version of ${{vars.pypi-package}}
    dependencies:
      provider-priority: ${{range.value}}
      provides:
        - py3-${{vars.pypi-package}}
      runtime:
        - py${{range.key}}-cryptography
        - py${{range.key}}-typing-extensions
    pipeline:
      - uses: py/pip-build-install
        with:
          python: python${{range.key}}
      - uses: strip
    test:
      environment:
        contents:
          packages:
            - py${{range.key}}-cryptography
            - py${{range.key}}-typing-extensions
            - python-${{range.key}}
      pipeline:
        - uses: python/import
          with:
            python: python${{range.key}}
            imports: |
              import ${{vars.pypi-package}}
        - name: Test basic AsyncSSH functionality
          uses: test/daemon-check-output
          with:
            start: |
              python${{range.key}} connection.py
            timeout: 10
            expected_output: |
              SSH command result
              hi
            error_strings: |
              Client error:
              Traceback
              Exception
              Error:

  - name: py3-supported-${{vars.pypi-package}}
    description: meta package providing ${{vars.pypi-package}} for supported python versions.
    dependencies:
      runtime:
        - py3.10-${{vars.pypi-package}}
        - py3.11-${{vars.pypi-package}}
        - py3.12-${{vars.pypi-package}}
        - py3.13-${{vars.pypi-package}}
    test:
      environment:
        contents:
          packages:
            - py3.10-cryptography
            - py3.10-typing-extensions
      pipeline:
        - uses: python/import
          with:
            python: python3.10
            imports: |
              import ${{vars.pypi-package}}

update:
  enabled: true
  github:
    identifier: ronf/asyncssh
    strip-prefix: v
    use-tag: true
